### PR TITLE
Registry perf

### PR DESCRIPTION
**Pull request by ludoo0d0a**
This PR refers to issue #370 

After indexing strings, search filter is simpler, more efficient, so faster (10x may be)

Todo : 
- keep time tracking on search in log ?

### CODE REVIEW

File: lib/widgets/common/base/gcw_textfield.dart [hunk 1]
```diff
@@ -78,31 +80,33 @@ class _GCWTextFieldState extends State<GCWTextField> {
...
-                        child: Container(
-                          child: Icon(
-                            Icons.clear,
-                            color: colors.mainFont(),
-                          ),
-                          padding: EdgeInsets.only(right: 5, top: 5, bottom: 5),
-                        ),
-                        onTap: () {
-                          if (widget.controller != null) widget.controller.clear();
+                suffixIcon: widget.clearIcon
```
Comment (S-Man42) [inline lib/widgets/common/base/gcw_textfield.dart:83]: Why did you remove the clear button? I use it regularly...

File: lib/widgets/main_view.dart [hunk 2]
```diff
@@ -369,9 +373,10 @@ class _MainViewState extends State<MainView> {
       ].contains(className(element.tool));
     }).toList();
 
-    _toolList.sort((a, b) {
-      return a.toolName.toLowerCase().compareTo(b.toolName.toLowerCase());
-    });
+    // already sorted in registry
```
Comment (S-Man42) [inline lib/widgets/main_view.dart:376]: Please remove code which is not useful anymore

File: lib/widgets/main_view.dart [hunk 3]
```diff
@@ -458,28 +466,29 @@ class _MainViewState extends State<MainView> {
...
-    list = list.where((tool) {
-      searchstring = tool.searchStrings.join(' ').toLowerCase();
-      if (searchstring == null || searchstring.length == 0) return false;
-
-      var found = true;
+    Stopwatch stopwatch = Stopwatch()..start();
+    List<String> _queryTexts = splitWords(removeAccents(_searchText.toLowerCase()));
 
+    var list = Registry.indexedTools.where((tool) {
+   // var list = Registry.toolList.where((tool) {
```
Comment (S-Man42) [inline lib/widgets/main_view.dart:473]: Please remove commented code. :)

File: lib/widgets/main_view.dart [hunk 4]
```diff
@@ -185,7 +185,11 @@ class _MainViewState extends State<MainView> {
 
     _searchController.addListener(() {
       setState(() {
-        _searchText = _searchController.text.isEmpty ? '' : _searchController.text;
+        if (_searchController.text.isEmpty){
+          _searchText = '';
+        }else if (_searchText!= _searchController.text) {
```
Comment (S-Man42) [inline lib/widgets/main_view.dart:190]: Did you format the code?. The missing spaces look strange ;)

**Comment by S-Man42** [comment]
When I'd merge your code, it seems that some files would change, i.e. the pubspec.yaml (which is on Build 95 in your branch, instead of 99). Please merge or rebase your current branch. I just saw this one change by accident... I can imagine, some other points (registry, searchstrings, ...) might change to a earlier version as well.

File: lib/widgets/common/base/gcw_textfield.dart (hunk 1 above)
Comment (ludoo0d0a) [reply lib/widgets/common/base/gcw_textfield.dart:83]: there a 2 clear buttons in search bar. I add option to remove only this one.

File: lib/widgets/main_view.dart (hunk 4 above)
Comment (ludoo0d0a) [reply lib/widgets/main_view.dart:190]: not a format issue, I add a check to not rebuild if string is not changed

File: lib/widgets/common/base/gcw_textfield.dart (hunk 1 above)
Comment (S-Man42) [reply lib/widgets/common/base/gcw_textfield.dart:83]: Sorry, I don't understand...

1.) Why are there 2 buttons? I have only one.
2.) If I try your code, the button is gone. 

So as I see, you remove it... What do I miss.

I currently see absolutely no reason, why someone should want to remove the clear button from the TextField widget, so I am struggling with this option very much.

What I am missing here? :)

Comment (ludoo0d0a) [reply lib/widgets/common/base/gcw_textfield.dart]: done

File: lib/widgets/main_view.dart (hunk 3 above)
Comment (ludoo0d0a) [reply lib/widgets/main_view.dart]: done

File: lib/widgets/main_view.dart (hunk 2 above)
Comment (ludoo0d0a) [reply lib/widgets/main_view.dart]: done

**Comment by ludoo0d0a** [comment]
I rebased on 1.60 but I'm really not sure if this PR is good
I think I need to help to understand "git rebase"

**Comment by ludoo0d0a** [comment]
see new PR properly rebased
https://github.com/S-Man42/GCWizard/compare/1.6.0...ludoo0d0a:registry_perf2?expand=1

**Comment by ludoo0d0a** [comment]
abandoned in favor of https://github.com/S-Man42/GCWizard/pull/538
